.github/workflows/markdown-lint.yml

```diff
@@ -31,13 +31,34 @@ jobs:
           yamllint --version || { printf "%s\n" "${ERR_MSG_LINE}"; exit 1; }
 
       - name: Lint Workflow YAML
+        if: ${{ success() }}
         run: |
-          yamllint ${{ vars.YAML_ARGS }} .github/workflows/markdown-lint.yml
+          YAML_ARGS="${{ vars.YAML_ARGS }}"
+          if [[ "${YAML_ARGS}" =~ ^*[!a-zA-Z0-9\ \-\.]*$ ]]; then
+              ERR_MSG_STUB="Invalid YAML_ARGS: contains unauthorized characters!"
+              ERR_LOC="file=.github/workflows/markdown-lint.yml,line=37,endLine=42,"
+              printf "::error %s%s\n" "${ERR_LOC}" "title=VALIDATION_ERROR::ERROR ${ERR_MSG_STUB}"
+              exit 1
+          else
+              printf "::notice file=.github/workflows/markdown-lint.yml,%s %s\n" \
+                "line=36,endLine=47,title=Validation::Validating workflow with args:" \
+                "${YAML_ARGS}"
+              read -ra ARGS <<< "${YAML_ARGS}"
+              yamllint "${ARGS[@]}" .github/workflows/markdown-lint.yml
+          fi
 
       - name: Install NPM Dependencies
         run: |
-          sudo npm install -g markdownlint
-          sudo npm install -g markdownlint-cli
+          # npm install
+          ERR_MSG="NPM package installation failed"
+          ERR_LOC_1="file=.github/workflows/markdown-lint.yml,line=56,endLine=56"
+          ERR_LOC_2="file=package.json,line=2,endLine=4"
+          if ! NPM_ERROR=$(npm install 2>&1); then
+          for LOC in "${ERR_LOC_1}" "${ERR_LOC_2}"; do
+            printf "::error %s,title=INSTALL_ERROR::ERROR %s: %s\n" \
+              "${LOC}" "${ERR_MSG}" "${NPM_ERROR}"
+          done
+          fi
```

Comment:
@codrabbitai, This change almost seems like it was overkill now, it caused more churn to debug than expected and does not feel better overall. Let's re-analyze (as a post-mortem) the changes to "Install NPM Dependencies" step in .github/workflows/markdown-lint.yml, considering the impacts to metrics like: Code-Size, Maintainability, Security, Stability, Complexity, Readability, in addition to other relevant metrics, (of-course, as a post mortem, include both positive and negative impacts of each metric). Thanks in advance.